templates/firm/show.json.twig

```diff
@@ -0,0 +1,61 @@
+{
+    "@context": "https://schema.org",
+    "@graph": [
+        {
+            "@type": "Organization"
+            {% if firm.name %},"name": "{{firm.name}}" {% endif %}
+            {% if firm.startDate %}, "foundingDate": "{{ firm.startDate }}" {% endif %}
+            {% if firm.endDate %} ,"dissolutionDate": "{{ firm.endDate }}" {% endif %},
+            "@id": "{{ url('firm_show', {'id':firm.id}) }}",
+            "address": {
+                "@type": "PostalAddress"
+                {% if firm.city %},"addressLocality": "{{ firm.city }}" {% endif %}
+                {% if firm.streetAddress %}, "streetAddress": "{{ firm.streetAddress }}" {% endif %}
+            }
+        }
+        {% for titleFirmRole in firm.getTitleFirmRoles("Publisher") %}
+            {% if titleFirmRole.getFirmrole.getName == "Publisher" %}
+                {% set book = titleFirmRole.title %}
+                {% if book %}
+                ,{
+                    "@type": "Book",
+                    "@id": "{{ url('title_show', {'id':book.id}) }}",
+                    "name": "{{book.title}}"
+                    {% if book.pubdate %}, "datePublished": "{{ book.pubdate }}" {% endif %}
+                    {% if book.locationOfPrinting %},
+                        "locationCreated": {
+                            "@type": "Place",
```

Comment:
Can you add the location's @id here? In the location table it's called geonameid instead of id.